# Autogen Installation


1. Create a venv
2. Create a requirements.txt
3. install autogen-agentchat autogen-core autogen-ext

# First Autogen Agent


In [1]:
from autogen_agentchat.agents import AssistantAgent
from autogen_ext.models.openai import OpenAIChatCompletionClient
from dotenv import load_dotenv
import os

load_dotenv()

api_key = os.getenv('OPENAI_API_KEY')

# Connecting to the Model

In [2]:
model_client = OpenAIChatCompletionClient(model='gpt-4o',api_key=api_key)


In [3]:
agent_1 = AssistantAgent(name='my_assistant',model_client=model_client)

In [4]:
result = await agent_1.run(task='Tell me something about you?')

In [5]:
print(result)

messages=[TextMessage(id='f1549ff4-d450-4ccf-9f44-76812fe6f45b', source='user', models_usage=None, metadata={}, created_at=datetime.datetime(2025, 7, 23, 22, 52, 10, 688016, tzinfo=datetime.timezone.utc), content='Tell me something about you?', type='TextMessage'), TextMessage(id='064c0387-ab70-425a-962a-4ebdf568b776', source='my_assistant', models_usage=RequestUsage(prompt_tokens=42, completion_tokens=93), metadata={}, created_at=datetime.datetime(2025, 7, 23, 22, 52, 13, 679892, tzinfo=datetime.timezone.utc), content="I am an AI assistant designed to help users by answering questions, providing information, and completing various tasks using a range of tools. I draw upon a vast amount of knowledge and datasets to perform these functions effectively. As I am not sentient, I do not possess feelings, thoughts, or personal experiences, but I'm here to assist you as best as I can. If there's any specific information or task you would like help with, feel free to let me know!", type='TextM

In [6]:
result2 = await agent_1.run(task='What was my last question?')

In [9]:
result2

TaskResult(messages=[TextMessage(id='1fcf6a02-3917-433a-be91-97bf6a7740e6', source='user', models_usage=None, metadata={}, created_at=datetime.datetime(2025, 7, 12, 2, 16, 11, 991206, tzinfo=datetime.timezone.utc), content='What was my last question?', type='TextMessage'), TextMessage(id='a515c43a-d1bd-423f-83f8-652a93ffcde0', source='my_assistant', models_usage=RequestUsage(prompt_tokens=149, completion_tokens=53), metadata={}, created_at=datetime.datetime(2025, 7, 12, 2, 16, 13, 376071, tzinfo=datetime.timezone.utc), content="I'm sorry, but I don't have the ability to remember past interactions or previous questions you've asked. Each interaction with me is independent, and I don't retain any information from prior conversations. If there's anything you'd like to ask or discuss, feel free to let me know!", type='TextMessage')], stop_reason=None)

# Code

In [6]:
from autogen_agentchat.agents import AssistantAgent
from autogen_ext.models.openai import OpenAIChatCompletionClient
from dotenv import load_dotenv
import os

# Load API Key
load_dotenv()
api_key = os.getenv('OPENAI_API_KEY')

# Connect to GPT-4o
model_client = OpenAIChatCompletionClient(
    model='gpt-4o',
    api_key=api_key
)

# Create Agent
agent_1 = AssistantAgent(
    name='my_assistant',
    model_client=model_client
)

# Run Tasks
result = await agent_1.run(task='What are the possibilities that AI is misused')
print(result)


messages=[TextMessage(id='ecbb2870-a613-42e0-b37a-08a405622e75', source='user', models_usage=None, metadata={}, created_at=datetime.datetime(2025, 7, 18, 22, 11, 27, 411808, tzinfo=datetime.timezone.utc), content='What are the possibilities that AI is misused', type='TextMessage'), TextMessage(id='6371aa21-a9c3-4d5b-8ea2-f27d0e984294', source='my_assistant', models_usage=RequestUsage(prompt_tokens=45, completion_tokens=476), metadata={}, created_at=datetime.datetime(2025, 7, 18, 22, 11, 34, 961386, tzinfo=datetime.timezone.utc), content="The potential misuse of artificial intelligence (AI) arises from its capacity to amplify human capabilities, both positive and negative. Here are several scenarios where AI could be misused:\n\n1. **Surveillance and Privacy Violations**:\n   - AI-powered surveillance systems can be used to monitor individuals without their consent, leading to privacy infringements and potential abuse by governments or corporations.\n\n2. **Deepfakes and Disinformation*

messages=[TextMessage(id='ecbb2870-a613-42e0-b37a-08a405622e75', source='user', models_usage=None, metadata={}, created_at=datetime.datetime(2025, 7, 18, 22, 11, 27, 411808, tzinfo=datetime.timezone.utc), content='What are the possibilities that AI is misused', type='TextMessage'), TextMessage(id='6371aa21-a9c3-4d5b-8ea2-f27d0e984294', source='my_assistant', models_usage=RequestUsage(prompt_tokens=45, completion_tokens=476), metadata={}, created_at=datetime.datetime(2025, 7, 18, 22, 11, 34, 961386, tzinfo=datetime.timezone.utc), 

content="The potential misuse of artificial intelligence (AI) arises from its capacity to amplify human capabilities, both positive and negative. Here are several scenarios where AI could be misused:\n\n1. **Surveillance and Privacy Violations**:\n   - AI-powered surveillance systems can be used to monitor individuals without their consent, leading to privacy infringements and potential abuse by governments or corporations.\n\n2. **Deepfakes and Disinformation**:\n   - AI can generate highly convincing fake videos, audio, or text (deepfakes), which can be used to spread misinformation, manipulate public opinion, or defame individuals.\n\n3. **Autonomous Weapons**:\n   - AI systems can be integrated into military applications, creating autonomous weapons that can make life-and-death decisions without human intervention, raising ethical and accountability issues.\n\n4. **Bias and Discrimination**:\n   - If AI systems are trained on biased data, they can perpetuate or even exacerbate these biases, leading to unfair treatment of individuals based on race, gender, or other characteristics.\n\n5. **Cybersecurity Threats**:\n   - AI can be used to enhance cyberattacks, making them more sophisticated and harder to detect, posing significant risks to data security and privacy.\n\n6. **Economic Displacement**:\n   - AI-driven automation can lead to job displacement, exacerbating economic inequalities and societal tensions if not managed properly.\n\n7. **Manipulation and Behavior Control**:\n   - AI could be used to manipulate consumer behavior or social dynamics in unethical ways, such as through targeted advertising or nudging, without the users' awareness.\n\n8. **Intellectual Property Theft**:\n   - AI algorithms can be used to clone or steal protected intellectual properties like patented algorithms or proprietary databases.\n\n9. **Fraud and Identity Theft**:\n   - AI can be employed to create sophisticated phishing schemes or to generate fake identities, complicating fraud detection.\n\n10. **Addiction and Mental Health Issues**:\n    - AI algorithms, particularly in social media, can be designed to maximize user engagement at the expense of mental well-being, leading to addiction or mental health struggles.\n\nPreventing the misuse of AI involves implementing robust ethical guidelines, regulatory frameworks, and oversight mechanisms to ensure that AI technologies are developed and deployed responsibly. It also requires transparency, accountability, and a multidisciplinary approach to address the complexities involved. \n\nTERMINATE", type='TextMessage')] stop_reason=None